Build failure: evdi

### Steps To Reproduce

Steps to reproduce the behavior:
1. Create nixos configuration with `boot.kernelPackages = pkgs.linuxPackages_6_1;`
2. Include option: `services.xserver.videoDrivers = [ "displaylink" "modesetting" ];`
3. `nixos-rebuild build`
### Build log

```
(truncated build log)

...
build/source/module/evdi_fb.c: In function 'evdifb_create':
/build/source/module/evdi_fb.c:408:23: error: 'struct drm_fb_helper' has no member named 'info'
  408 |         efbdev->helper.info = info;
      |                       ^
/build/source/module/evdi_fb.c: In function 'evdi_fbdev_destroy':
/build/source/module/evdi_fb.c:468:27: error: 'struct drm_fb_helper' has no member named 'info'
  468 |         if (efbdev->helper.info) {
      |                           ^
/build/source/module/evdi_fb.c:469:38: error: 'struct drm_fb_helper' has no member named 'info'
  469 |                 info = efbdev->helper.info;
      |                                      ^
/build/source/module/evdi_fb.c: In function 'evdi_fbdev_unplug':
/build/source/module/evdi_fb.c:560:27: error: 'struct drm_fb_helper' has no member named 'info'
  560 |         if (efbdev->helper.info) {
      |                           ^
/build/source/module/evdi_fb.c:563:38: error: 'struct drm_fb_helper' has no member named 'info'
  563 |                 info = efbdev->helper.info;

...

make[2]: *** [/nix/store/yyp3x1hyllmm490b43h4zqyrbyzxc28v-linux-6.1.111-dev/lib/modules/6.1.111/source/Makefile:2009: /build/source/module] Error 2
make[1]: *** [Makefile:87: module] Error 2
make[1]: Leaving directory '/build/source/module'
make: *** [Makefile:13: module] Error 2

```

### Additional context

Without pinning kernel version to 6.1 it's all good (kernel 6.6)

### Notify maintainers

@abbradar 
<!--
Please @ people who are in the `meta.maintainers` list of the offending package or module.
If in doubt, check `git blame` for whoever last touched something.
-->

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.1.102, NixOS, 24.11 (Vicuna), 24.11.20240921.9357f4f`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.7`
 - nixpkgs: `/nix/store/y6205wq8hxvpqvl8l9d1n9xah01kg0lq-source`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/os-specific/linux/evdi/default.nix
+++ b/pkgs/os-specific/linux/evdi/default.nix
@@ -2,6 +2,7 @@
   lib,
   stdenv,
   fetchFromGitHub,
+  fetchpatch,
   kernel,
   libdrm,
   python3,
@@ -16,14 +17,21 @@ let
 in
 stdenv.mkDerivation (finalAttrs: {
   pname = "evdi";
-  version = "1.14.7";
+  version = "1.14.7-unstable-2024-11-30";
 
   src = fetchFromGitHub {
     owner = "DisplayLink";
     repo = "evdi";
-    rev = "refs/tags/v${finalAttrs.version}";
-    hash = "sha256-z3GawjaokbmmUC1LihwGSnF3tUp9n/FO+kDiWvBq+mY=";
+    rev = "59a3a864f7476cd61d9c65bfd012d1e9ed90e2b1";
+    hash = "sha256-0xEh0Tb5QFReW5lXO/Mb3gn1z87+baR8Tix+dQjUZMw=";
   };
+
+  patches = [
+    (fetchpatch {
+      url = "https://github.com/DisplayLink/evdi/commit/e41240cf62d7188643bc95e5d69e1c4cfa6ddb84.patch?full_index=1";
+      hash = "sha256-6V3QJZMAhXqfGLW2eWkIzJnOdBPvLLNVzg6DW1M3IaA=";
+    })
+  ];
 
   env.NIX_CFLAGS_COMPILE = toString [
     "-Wno-error"
@@ -60,7 +68,7 @@ stdenv.mkDerivation (finalAttrs: {
   enableParallelBuilding = true;
 
   meta = {
-    broken = kernel.kernelOlder "4.19" || kernel.kernelAtLeast "6.12";
+    broken = kernel.kernelOlder "4.19";
     changelog = "https://github.com/DisplayLink/evdi/releases/tag/v${finalAttrs.version}";
     description = "Extensible Virtual Display Interface";
     homepage = "https://www.displaylink.com/";

